gxa/src/test/java/uk/ac/ebi/atlas/experimentpage/differential/DifferentialPathwaysComparisonServiceTest.java

```diff
@@ -0,0 +1,247 @@
+    private String[] c2s9 = new String[]{"R-MMU-2029480", "Fcgamma receptor (FCGR) dependent phagocytosis", "45", "0.535169358108118", "0.0436946551249873"};
+    private String[] c2s10 = new String[]{"R-MMU-5654741", "Signaling by FGFR3", "45", "0.535169358108118", "0.0436946551249873"};
+    private String[] c2s11 = new String[]{"R-MMU-1483206", "Apoptosis", "45", "0.535169358108118", "0.0436946551249873"};
+
+
+    @Before
+    public void setUp() throws Exception {
+        MockDataFileHub mockDataFileHub = new MockDataFileHub();
+
+        String accession = "E-GEOD-54351";
+
+        mockDataFileHub.addReactomePathwaysFile(accession, c1.getId(),
+                Lists.newArrayList(c1s1, c1s2, c1s3, c1s4, c1s5, c1s6, c1s7, c1s8, c1s9, c1s10, c1s11));
+        mockDataFileHub.addReactomePathwaysFile(accession, c2.getId(),
+                Lists.newArrayList(c2s1, c2s2, c2s3, c2s4, c2s5, c2s6, c2s7, c2s8, c2s9, c2s10, c2s11));
+
+        this.subject = new DifferentialPathwaysComparisonService<>(mockDataFileHub, mockConfigurationTrader,
+                mockDifferentialRequestContextFactory, mockDifferentialProfilesHeatMap);
+
+        when(mockDifferentialExperiment.getAccession()).thenReturn(accession);
+        when(mockConfigurationTrader.getMicroarrayExperimentConfiguration(accession)).thenReturn(mockMicroarrayExperimentConfiguration);
+        when(mockConfigurationTrader.getMicroarrayExperimentConfiguration(accession).getContrasts()).thenReturn(Lists.newArrayList(c1,c2));
+
+        setUpDataForTest();
+    }
+
+    @Test
+    public void testGetPathwaysListPerComparison() throws Exception {
+
+        Map comparisonPathwaysMap = subject.getPathwaysListPerComparison(mockDifferentialExperiment);
+
+        assertNotNull(comparisonPathwaysMap);
+        assertThat(comparisonPathwaysMap.size(), is(2));
+        assertThat(comparisonPathwaysMap.get(c1), is(new ArrayList<>()));
+
+        List<String> expectedList = Arrays.asList("R-MMU-416482", "R-MMU-397014", "R-MMU-2132295", "R-MMU-193648", "R-MMU-373760",
+                "R-MMU-5620924", "R-MMU-5654741", "R-MMU-199418", "R-MMU-2029480", "R-MMU-1483206");
+        assertThat(comparisonPathwaysMap.get(c2), is(expectedList));
+
+    }
+
+    @Test
+    public void testConstructPathwaysByComparison() throws Exception {
+
+        Map multimap = subject.constructPathwaysByComparison(mockDifferentialExperiment, mockDifferentialRequestPreferences);
+        assertNotNull(multimap);
+        assertThat(multimap.size(), is(1));
+
+        //Contrast 2
+        HashMultimap values = (HashMultimap) multimap.get(c2);
+        Map map = values.asMap();
+
+        Set<String> expectedPathways = Sets.newHashSet("R-MMU-416482", "R-MMU-397014", "R-MMU-199418", "R-MMU-193648", "R-MMU-1483206");
+        assertThat(map.keySet(), is(expectedPathways));
+
+        Collection<Pair<String, DifferentialExpression>> pairCollection1 = (Collection<Pair<String, DifferentialExpression>>) map.get("R-MMU-416482");
```

Comment:
This looks like it took lots of effort, and maybe it wasn't required.
For these kinds of tests I often do
```
assertThat(result, is(<very big object>))
```

When the result is JSON, I type in this first:
```
assertThat(result, is(new Gson().fromJson("{}", JsonObject.class)));
```
This of course fails, because the result is not "{}", and the failing test tells me what it was instead. Then I take what the result actually was and paste it in. :)

Reply:
Well, actually I did it this way because I got one case that the expected result was different and needed to be sure that the code was returning the "expectedPathways". I know it has lot of effort and that is why it took me a while

Reply:
Yea like I said you've created lots of work for yourself.